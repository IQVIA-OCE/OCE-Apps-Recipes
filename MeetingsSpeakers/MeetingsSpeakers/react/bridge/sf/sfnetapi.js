// @flow

import { NativeModules } from 'react-native';
const SFNetReactBridge = NativeModules.SFNetReactBridge;
import { environment } from '../EnvironmentData/EnvironmentData.native';
import { exec as forceExec } from './react.force.common';

type SuccessCallback = (value: any) => void;
type ErrorCallback = (error: Error | null) => void;

class SFNetAPI {
  apiVersion: string = environment.sfApiVersion();
  usePromises: boolean;

  /**
   * Set apiVersion to be used
   */
  setApiVersion(version: string) {
    this.apiVersion = version;
  }

  /**
   * Return apiVersion used
   */
  getApiVersion(): string {
    return this.apiVersion;
  }

  /**
   * Send arbitray force.com request
   */
  sendRequest(
    endPoint: string,
    path: string,
    successCB: SuccessCallback,
    errorCB: ErrorCallback,
    method: ?string,
    payload: ?any,
    headerParams: ?any,
    fileParams: ?any,
    returnBinary: ?any,
    doesNotRequireAuthentication: ?any
  ): void {
    method = method || 'GET';
    payload = payload || {};
    headerParams = headerParams || {};
    fileParams = fileParams || {}; // File params expected to be of the form: {<fileParamNameInPost>: {fileMimeType:<someMimeType>, fileUrl:<fileUrl>, fileName:<fileNameForPost>}}
    returnBinary = !!returnBinary;
    doesNotRequireAuthentication = !!doesNotRequireAuthentication;
    const args = {
      endPoint,
      path,
      method,
      queryParams: payload,
      headerParams,
      fileParams,
      returnBinary,
      doesNotRequireAuthentication,
    };
    forceExec(
      'SFNetReactBridge',
      SFNetReactBridge,
      successCB,
      errorCB,
      'sendRequest',
      args
    );
  }

  /*
   * Lists summary information about each Salesforce.com version currently
   * available, including the version, label, and a link to each version's
   * root.
   * @param callback function to which response will be passed
   * @param [error=null] function called in case of error
   */
  //export const versions = (callback, error) => sendRequest('/services/data', '/', callback, error);

  /*
   * Lists available resources for the client's API version, including
   * resource name and URI.
   * @param callback function to which response will be passed
   * @param [error=null] function called in case of error
   */
  //export const resources = (callback, error) => sendRequest('/services/data', `/${apiVersion}/`, callback, error);

  /*
   * Lists the available objects and their metadata for your organization's
   * data.
   * @param callback function to which response will be passed
   * @param [error=null] function called in case of error
   */
  describeGlobal(callback: SuccessCallback, error: ErrorCallback) {
    this.sendRequest(
      '/services/data',
      `/${this.apiVersion}/sobjects/`,
      callback,
      error
    );
  }

  /*
   * Describes the individual metadata for the specified object.
   * @param objtype object type; e.g. "Account"
   * @param callback function to which response will be passed
   * @param [error=null] function called in case of error
   */
  metadata(objtype: string, callback: SuccessCallback, error: ErrorCallback) {
    if (this.usePromises) {
    }
    this.sendRequest(
      '/services/data',
      `/${this.apiVersion}/sobjects/${objtype}/`,
      callback,
      error
    );
  }

  /*
   * Completely describes the individual metadata at all levels for the
   * specified object.
   * @param objtype object type; e.g. "Account"
   * @param callback function to which response will be passed
   * @param [error=null] function called in case of error
   */
  describe(objtype: string, callback: SuccessCallback, error: ErrorCallback) {
    this.sendRequest(
      '/services/data',
      `/${this.apiVersion}/sobjects/${objtype}/describe/`,
      callback,
      error
    );
  }

  /*
   * Fetches the layout configuration for a particular sobject type and record type id.
   * @param objtype object type; e.g. "Account"
   * @param (Optional) recordTypeId Id of the layout's associated record type
   * @param callback function to which response will be passed
   * @param [error=null] function called in case of error
   */
  describeLayout(
    objtype: string,
    recordTypeId: string,
    callback: SuccessCallback,
    error: ErrorCallback
  ): void {
    recordTypeId = recordTypeId ? recordTypeId : '';
    this.sendRequest(
      '/services/data',
      `/${this.apiVersion}/sobjects/${objtype}/describe/layouts/${recordTypeId}`,
      callback,
      error
    );
  }

  /*
   * Creates a new record of the given type.
   * @param objtype object type; e.g. "Account"
   * @param fields an object containing initial field names and values for
   *               the record, e.g. {:Name "salesforce.com", :TickerSymbol
   *               "CRM"}
   * @param callback function to which response will be passed
   * @param [error=null] function called in case of error
   */
  create(
    objtype: string,
    fields: any,
    callback: SuccessCallback,
    error: ErrorCallback
  ): void {
    this.sendRequest(
      '/services/data',
      `/${this.apiVersion}/sobjects/${objtype}/`,
      callback,
      error,
      'POST',
      fields
    );
  }

  /*
   * Retrieves field values for a record of the given type.
   * @param objtype object type; e.g. "Account"
   * @param id the record's object ID
   * @param [fields=null] optional comma-separated list of fields for which
   *               to return values; e.g. Name,Industry,TickerSymbol
   * @param callback function to which response will be passed
   * @param [error=null] function called in case of error
   */
  retrieve(
    objtype: string,
    id: string,
    fieldlist: any,
    callback: SuccessCallback,
    error: ErrorCallback
  ): void {
    if (arguments.length == 4) {
      error = callback;
      callback = fieldlist;
      fieldlist = null;
    }
    const fields = fieldlist ? { fields: fieldlist } : null;
    return this.sendRequest(
      '/services/data',
      `/${this.apiVersion}/sobjects/${objtype}/${id}`,
      callback,
      error,
      'GET',
      fields
    );
  }

  /*
   * Upsert - creates or updates record of the given type, based on the
   * given external Id.
   * @param objtype object type; e.g. "Account"
   * @param externalIdField external ID field name; e.g. "accountMaster__c"
   * @param externalId the record's external ID value
   * @param fields an object containing field names and values for
   *               the record, e.g. {:Name "salesforce.com", :TickerSymbol
   *               "CRM"}
   * @param callback function to which response will be passed
   * @param [error=null] function called in case of error
   */
  upsert(
    objtype: string,
    externalIdField: string,
    externalId: string,
    fields: any,
    callback: SuccessCallback,
    error: ErrorCallback
  ): void {
    this.sendRequest(
      '/services/data',
      `/${this.apiVersion}/sobjects/${objtype}/${externalIdField}/${externalId}`,
      callback,
      error,
      externalId ? 'PATCH' : 'POST',
      fields
    );
  }
  /*
   * Updates field values on a record of the given type.
   * @param objtype object type; e.g. "Account"
   * @param id the record's object ID
   * @param fields an object containing initial field names and values for
   *               the record, e.g. {:Name "salesforce.com", :TickerSymbol
   *               "CRM"}
   * @param callback function to which response will be passed
   * @param [error=null] function called in case of error
   */
  update(
    objtype: string,
    id: string,
    fields: any,
    callback: SuccessCallback,
    error: ErrorCallback
  ): void {
    this.sendRequest(
      '/services/data',
      `/${this.apiVersion}/sobjects/${objtype}/${id}`,
      callback,
      error,
      'PATCH',
      fields
    );
  }

  /*
   * Deletes a record of the given type. Unfortunately, 'delete' is a
   * reserved word in JavaScript.
   * @param objtype object type; e.g. "Account"
   * @param id the record's object ID
   * @param callback function to which response will be passed
   * @param [error=null] function called in case of error
   */
  del(
    objtype: string,
    id: string,
    callback: SuccessCallback,
    error: ErrorCallback
  ): void {
    this.sendRequest(
      '/services/data',
      `/${this.apiVersion}/sobjects/${objtype}/${id}`,
      callback,
      error,
      'DELETE'
    );
  }

  /*
   * Executes the specified SOQL query.
   * @param soql a string containing the query to execute - e.g. "SELECT Id,
   *             Name from Account ORDER BY Name LIMIT 20"
   * @param callback function to which response will be passed
   * @param [error=null] function called in case of error
   */
  query(soql: string, callback: SuccessCallback, error: ErrorCallback): void {
    this.sendRequest(
      '/services/data',
      `/${this.apiVersion}/query`,
      callback,
      error,
      'GET',
      { q: soql }
    );
  }

  /*
   * Queries the next set of records based on pagination.
   * <p>This should be used if performing a query that retrieves more than can be returned
   * in accordance with http://www.salesforce.com/us/developer/docs/api_rest/Content/dome_query.htm</p>
   * @param url - the url retrieved from nextRecordsUrl or prevRecordsUrl
   * @param callback function to which response will be passed
   * @param [error=null] function called in case of error
   */
  queryMore(
    url: string,
    callback: SuccessCallback,
    error: ErrorCallback
  ): void {
    // $FlowIgnore
    const pathFromUrl = url.match(/https:\/\/[^/]*(.*)/)[1];
    this.sendRequest('', pathFromUrl, callback, error);
  }

  /*
   * Executes the specified SOSL search.
   * @param sosl a string containing the search to execute - e.g. "FIND
   *             {needle}"
   * @param callback function to which response will be passed
   * @param [error=null] function called in case of error
   */
  search(sosl: string, callback: SuccessCallback, error: ErrorCallback): void {
    this.sendRequest(
      '/services/data',
      `/${this.apiVersion}/search`,
      callback,
      error,
      'GET',
      { q: sosl }
    );
  }

  /*
   * Merges few requests to one composite request
   * @param requests Array of subrequests to execute.
   * @param refIds Array of reference id for the requests (should have the same number of element than requests)
   * @param allOrNone Specifies what to do when an error occurs while processing a subrequest.
   * @see https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/resources_composite_composite.htm
   * @param callback function to which response will be passed
   * @param [error=null] function called in case of error
   */
  composite(
    requests: Array<any>,
    refIds: Array<any>,
    allOrNone: any,
    callback: SuccessCallback,
    error: ErrorCallback
  ): void {
    const args = { requests, refIds, allOrNone };
    forceExec(
      'SFNetReactBridge',
      SFNetReactBridge,
      callback,
      error,
      'compositeRequests',
      args
    );
  }

  /*
   * Runs apex request
   * @param method sets the http method
   * @param endPoint sets the path ending
   * @param namespace sets the namespace
   * @param params sets the additional query params
   * @param callback function to which response will be passed
   * @param [error=null] function called in case of error
   */
  apexRest(
    method: string,
    endPoint: string,
    namespace: string,
    params: any,
    callback: SuccessCallback,
    error: ErrorCallback
  ): void {
    const args = { method, endPoint, namespace, params };
    forceExec(
      'SFNetReactBridge',
      SFNetReactBridge,
      callback,
      error,
      'apexRestWithMethod',
      args
    );
  }

  /*
   * Gets the report by id
   * @param reportId sets the id of report
   * @param callback function to which response will be passed
   * @param [error=null] function called in case of error
   */

  report(
    reportId: string,
    callback: SuccessCallback,
    error: ErrorCallback
  ): void {
    this.sendRequest(
      '/services/data',
      `/${this.apiVersion}/analytics/reports/${reportId}`,
      callback,
      error,
      'GET'
    );
  }
}

export const sfNetAPI: SFNetAPI = new SFNetAPI();
