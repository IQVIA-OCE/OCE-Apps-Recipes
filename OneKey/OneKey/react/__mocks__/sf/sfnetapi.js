import { NativeModules } from 'react-native';

NativeModules.SFNetReactBridge = {
  sendRequest: jest.fn(),
};

export default {
  apiVersion: '1',
  setApiVersion: jest.fn(),
  getApiVersion: jest.fn(),
  sendRequest: jest.fn(),
  describeGlobal: jest.fn(),
  metadata: jest.fn(),
  describe: jest.fn(),
  describeLayout: jest.fn(),
  create: jest.fn(),
  retrieve: jest.fn(),
  upsert: jest.fn(),
  update: jest.fn(),
  del: jest.fn(),
  query: jest.fn(),
  queryMore: jest.fn(),
  search: jest.fn(),
  composite: jest.fn(),
  apexRest: jest.fn(),
  report: jest.fn(),
}
