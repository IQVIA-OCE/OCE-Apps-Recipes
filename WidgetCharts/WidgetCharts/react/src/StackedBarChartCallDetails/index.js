import React from 'react';
import { View, Text, StyleSheet } from 'react-native';

import { sfNetAPI } from '../../bridge/sf/sfnetapi';

import {
  ActivityIndicator,
  Colors,
  BarChart,
  Button,
  IconButton,
} from 'apollo-react-native';
import { environment } from '../../bridge/EnvironmentData/EnvironmentData.native';
import { NavigationActions } from '../../bridge/Navigation/NavigationActions';
import { NEW_MODE } from '../../bridge/Navigation/NavigationModes';
import { navigator } from '../../bridge/Navigation/ScreenNavigator';
import moment from 'moment';
import { isIphone } from '../constants';

const colorScale = [
  '#32a2fb',
  '#91c9f6',
  '#6151f0',
  '#a9a0f1',
  '#fc1169',
  '#f77bab',
  '#b6b6b6',
];

export default class StackedBarChartCallDetails extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      chartData: [],
      loading: false,
      market: null,
      territory: environment.territory(),
      userId: environment.userID(),
    };
  }

  async componentDidMount() {
    try {
      const { chartData, minDate, maxDate } = await this.fetchCalldata();

      this.setState({
        chartData,
        minDate,
        maxDate,
      });
    } catch (error) {
      console.log(error);
    }
  }

  fetchCalldata = () => {
    const { territory } = this.state;
    const { recordId } = this.props;

    let filter = recordId
      ? `And OCE__Territory__c = '${territory.name}' And OCE__Account__c = '${recordId}'`
      : `And OCE__Territory__c = '${territory.name}'`;
    this.setState({ loading: true });

    return new Promise((resolve, reject) => {
      sfNetAPI.query(
        `SELECT OCE__CallType__c, OCE__CallDateTime__c FROM OCE__Call__c WHERE OCE__CallDateTime__c != null ${filter} ORDER BY OCE__CallDateTime__c`,
        data => {
          let chartData = [];
          let callDataByType = groupBy(data.records, 'OCE__CallType__c');
          let minDate;
          let maxDate;

          if (data.records.length) {
            minDate = `${data.records[0].OCE__CallDateTime__c.substring(
              0,
              8
            )}01`;
            maxDate = `${data.records[
              data.records.length - 1
            ].OCE__CallDateTime__c.substring(0, 8)}01`;
          }

          Object.keys(callDataByType).map(callType => {
            let groupedByDate = callDataByType[callType].reduce(
              (r, { OCE__CallDateTime__c }) => {
                if (OCE__CallDateTime__c) {
                  let formatedCallDateTime = `${OCE__CallDateTime__c.substring(
                    0,
                    8
                  )}01`;

                  if (!r[formatedCallDateTime])
                    r[formatedCallDateTime] = {
                      x: new Date(formatedCallDateTime),
                      y: 1,
                    };
                  else r[formatedCallDateTime].y++;
                }
                return r;
              },
              {}
            );
            const callValues = Object.keys(groupedByDate).map(
              key => groupedByDate[key]
            );

            chartData = [...chartData, { name: callType, data: callValues }];
          });

          this.setState({ loading: false });

          resolve({ chartData, minDate, maxDate });
        },
        error => {
          this.setState({ loading: false });
          console.log(error);
          reject(error);
        }
      );
    });
  };

  logACall(id) {
    const actions = NavigationActions.new({
      entity: 'OCE__Call__c',
      mode: NEW_MODE,
      parentID: id,
    });

    navigator.dispatch(actions).then(
      r => console.log(r),
      e => console.log(e)
    );
  }

  render() {
    const { chartData, minDate, maxDate, loading } = this.state;
    const { recordTypeId, connectionStatus } = this.props;
    let zoomDomain;

    if (minDate) {
      const zoomDomainMin = moment().utc(new Date(minDate)).subtract(1, 'M');
      const zoomDomainMax = moment().utc(new Date(minDate))
        .add(1, 'Y')
        .subtract(1, 'M');

      zoomDomain = { x: [new Date(zoomDomainMin), new Date(zoomDomainMax)] };
    }

    return (
      <View>
        <View
          style={[
            styles.topButtons,
            {
              position: isIphone ? 'relative' : 'absolute',
              top: isIphone ? 0 : -65,
              width: isIphone ? '100%' : 250,
              right: isIphone ? 20 : 0,
            },
          ]}
        >
          {connectionStatus != 'No Connection' ? (
            <View style={styles.refreshButton}>
              <IconButton
                icon="refresh"
                color="#2bb3fe"
                size={30}
                animated
                onPress={() => this.fetchCalldata()}
                disabled={loading}
              />
            </View>
          ) : null}
          <Button
            style={styles.buttonText}
            icon="phone"
            mode="outlined"
            onPress={() => this.logACall(recordTypeId)}
          >
            Call
          </Button>
        </View>

        {loading && (
          <View style={{ padding: 20 }}>
            <ActivityIndicator
              animating={true}
              color={Colors.blue700}
              style={{ paddingVertical: 10 }}
            />
          </View>
        )}

        {chartData.length && !loading ? (
          <BarChart
            data={chartData}
            colorScale={colorScale}
            height={250}
            allowZoom={false}
            allowPan={true}
            zoomDimension="x"
            zoomDomain={zoomDomain}
            scaleX={'time'}
            maxDomain={{ x: new Date('2022-01-01') }}
            minDomain={{ x: moment().utc(new Date(minDate)).subtract(1, 'M') }}
            tickFormatX={x => moment().utc(x).format('MMM, YY')}
            tickCount={12}
            hasAxis={true}
          />
        ) : null}

        {!chartData.length && !loading ? (
          <View style={{ alignSelf: 'center', paddingTop: 50 }}>
            <Text>No Data Found</Text>
          </View>
        ) : null}
      </View>
    );
  }
}

const groupBy = (items, key) =>
  items.reduce(
    (result, item) => ({
      ...result,
      [item[key]]: [...(result[item[key]] || []), item],
    }),
    {}
  );

const styles = StyleSheet.create({
  buttonText: {
    color: '#2bb3fe',
    fontSize: 16,
  },
  topButtons: {
    right: 0,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'flex-end',
  },
  refreshButton: {
    marginRight: 5,
  },
});
