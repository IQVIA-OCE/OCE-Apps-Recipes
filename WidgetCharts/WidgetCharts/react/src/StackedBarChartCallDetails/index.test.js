import React from 'react';
import StackedBarChartCallDetails from './index';
import renderer, { act } from 'react-test-renderer';
import { sfNetAPI } from '../../bridge/sf/sfnetapi';
import { Button } from 'apollo-react-native';
import moment from 'moment';

jest.mock('moment');


describe('LineChartTRXDetails', () => {
  beforeAll(() => {


    sfNetAPI.query = jest.fn()
      .mockImplementationOnce((query, cb) => {
        cb({
          records: [{
            OCE__CallDateTime__c: 'Tue May 1 2020 16:31:00 GMT+0300 (Eastern European Summer Time)'
          }]
        })
      })
      .mockImplementationOnce((query, cb, err) => {
        err({
          message: 'ERROR'
        })
      })
      .mockImplementationOnce((query, cb) => {
        cb({
          records: []
        })
      })
  });

  it('should render properly', async () => {
    let tree;

    await act(() => {
      tree = renderer.create(
        <StackedBarChartCallDetails />
      );
    });

    expect(tree.toJSON()).toMatchSnapshot();

  });


  it('should render with fetchDataError', () => {
    const tree = renderer.create(
      <StackedBarChartCallDetails />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render with no data', () => {
    const tree = renderer.create(
      <StackedBarChartCallDetails />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
