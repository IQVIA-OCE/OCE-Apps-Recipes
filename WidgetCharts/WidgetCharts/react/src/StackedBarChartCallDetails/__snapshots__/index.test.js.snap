// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineChartTRXDetails should render properly 1`] = `null`;

exports[`LineChartTRXDetails should render with fetchDataError 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "right": 0,
        },
        Object {
          "position": "relative",
          "right": 20,
          "top": 0,
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "marginRight": 5,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessibilityTraits="button"
        accessible={true}
        centered={true}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        onClick={[Function]}
        onLayout={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        pointerEvents="box-only"
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "margin": 6,
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 22.5,
                  "height": 45,
                  "width": 45,
                },
                false,
                undefined,
              ],
            ]
          }
        >
          <View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                  },
                  Object {
                    "height": 30,
                    "width": 30,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "transform": Array [
                      Object {
                        "rotate": "0deg",
                      },
                    ],
                  }
                }
              >
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  style={
                    Array [
                      Object {
                        "color": "#2bb3fe",
                        "fontSize": 30,
                      },
                      Array [
                        Object {
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "bottom": 0,
                "left": 0,
                "overflow": "hidden",
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "borderRadius": 0,
              },
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#0768FD",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 1,
          "color": "#2bb3fe",
          "elevation": 0,
          "fontSize": 16,
          "minWidth": 64,
          "opacity": 1,
          "width": "auto",
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessibilityTraits="button"
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onLayout={[Function]}
        onPressIn={[Function]}
        onPressOut={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        pointerEvents="box-only"
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Array [
                Object {
                  "borderRadius": 4,
                },
                undefined,
              ],
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "marginBottom": -1,
                  "marginLeft": 12,
                  "marginRight": -4,
                  "width": 16,
                }
              }
            >
              <Text
                accessibilityElementsHidden={true}
                allowFontScaling={false}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                style={
                  Array [
                    Object {
                      "color": "#0768FD",
                      "fontSize": 16,
                    },
                    Array [
                      Object {
                        "transform": Array [
                          Object {
                            "scaleX": 1,
                          },
                        ],
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                    ],
                    Object {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "letterSpacing": 1,
                      "textAlign": "center",
                    },
                    undefined,
                    false,
                    Object {
                      "color": "#0768FD",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": "500",
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                    },
                    Object {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Call
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "bottom": 0,
                "left": 0,
                "overflow": "hidden",
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "borderRadius": 0,
              },
            ]
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "paddingTop": 50,
      }
    }
  >
    <Text>
      No Data Found
    </Text>
  </View>
</View>
`;

exports[`LineChartTRXDetails should render with no data 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "right": 0,
        },
        Object {
          "position": "relative",
          "right": 20,
          "top": 0,
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "marginRight": 5,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessibilityTraits="button"
        accessible={true}
        centered={true}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        onClick={[Function]}
        onLayout={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        pointerEvents="box-only"
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "margin": 6,
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 22.5,
                  "height": 45,
                  "width": 45,
                },
                false,
                undefined,
              ],
            ]
          }
        >
          <View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                  },
                  Object {
                    "height": 30,
                    "width": 30,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "transform": Array [
                      Object {
                        "rotate": "0deg",
                      },
                    ],
                  }
                }
              >
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  style={
                    Array [
                      Object {
                        "color": "#2bb3fe",
                        "fontSize": 30,
                      },
                      Array [
                        Object {
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "bottom": 0,
                "left": 0,
                "overflow": "hidden",
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "borderRadius": 0,
              },
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#0768FD",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 1,
          "color": "#2bb3fe",
          "elevation": 0,
          "fontSize": 16,
          "minWidth": 64,
          "opacity": 1,
          "width": "auto",
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessibilityTraits="button"
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onLayout={[Function]}
        onPressIn={[Function]}
        onPressOut={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        pointerEvents="box-only"
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Array [
                Object {
                  "borderRadius": 4,
                },
                undefined,
              ],
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "marginBottom": -1,
                  "marginLeft": 12,
                  "marginRight": -4,
                  "width": 16,
                }
              }
            >
              <Text
                accessibilityElementsHidden={true}
                allowFontScaling={false}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                style={
                  Array [
                    Object {
                      "color": "#0768FD",
                      "fontSize": 16,
                    },
                    Array [
                      Object {
                        "transform": Array [
                          Object {
                            "scaleX": 1,
                          },
                        ],
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                    ],
                    Object {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "letterSpacing": 1,
                      "textAlign": "center",
                    },
                    undefined,
                    false,
                    Object {
                      "color": "#0768FD",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": "500",
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                    },
                    Object {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Call
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "bottom": 0,
                "left": 0,
                "overflow": "hidden",
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "borderRadius": 0,
              },
            ]
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "paddingTop": 50,
      }
    }
  >
    <Text>
      No Data Found
    </Text>
  </View>
</View>
`;
